ethers/contract.nim

```diff
@@ -292,9 +292,13 @@ proc subscribe*[E: Event](contract: Contract,
   let topic = topic($E, E.fieldTypes).toArray
   let filter = EventFilter(address: contract.address, topics: @[topic])
 
-  proc logHandler(log: Log) {.raises: [].} =
+  proc logHandler(logResult: ?!Log) {.raises: [].} =
+    without log ?= logResult, err:
+      handler(failure(e))
+      return
+
     if event =? E.decode(log.data, log.topics):
-      handler(event)
+      handler(success(event))
```

Comment:
Wouldn't you just want to pass in the result of `E.decode`? It returns `?!E` already, and that way decoding errors won't get swallowed.